overlays/haskell.nix

```diff
@@ -148,17 +151,20 @@ final: prev: {
             , pkg-def-extras ? []
             , modules ? []
             , extra-hackages ? []
-            }@args:
+            , compiler-nix-name ? null
+            }:
 
             let
-                pkg-def = excludeBootPackages plan-pkgs.pkgs;
-                # The compiler referenced in the stack config
-                compiler = (plan-pkgs.extras hackage).compiler or (pkg-def hackage).compiler;
-                patchesModule = ghcHackagePatches.${compiler.nix-name} or {};
+                compiler-nix-name' =
+                  if compiler-nix-name != null
+                    then compiler-nix-name
+                    else ((plan-pkgs.extras hackage).compiler or (plan-pkgs.pkgs hackage).compiler).nix-name;
```

Comment:
Should we have some kind of assertion that the compilers are at least the same version or something?